### ISSUE TITLE
Chapter exercise links using anchors #chapter-exercise-1 do not land on the correct scroll position because the exercise frame heights change after the page load

### ISSUE THREAD

**Issue by Mankro** [apluslms/a-plus#1028]
Using a toc-link to navigate to an exercise uses an anchor to the exercise element id, which seems to create a "data race" between the automatic scrolling and exercise views opening to full height. The web page does not end up on the correct point (at the exercise frame) since the exercises move around after they have settled on the correct height for their frames.

Solution 1:
record exercise height after loading and send it to server. Server adds element height to the html based on average of above information. Possible to fool the server with false data? 

Solution 2: 
JS updates location after loading exercise to update the scrolling position. If user creates scroll event, above logic is disabled.

----------------------------

I created this issue from the previous combined issue list (#438). The old description here is rather messy. Solution 1 sounds very strange. Solution 2 makes more sense. The view port could be set to the correct scroll position after the exercise frames have set their heights.

**Comment by EerikSaksi** [comment]
I'll do this if no one is assigned yet

**Comment by Mankro** [comment]
> I'll do this if no one is assigned yet

@EerikSaksi sure, go ahead!

**Comment by EerikSaksi** [comment]
https://user-images.githubusercontent.com/46625399/190600783-435af28f-9a32-4bc2-ab81-d388ee8f10ba.mp4

I either cannot replicate the issue locally or do not understand what the expected behaviour should be. This was recorded on Firefox but the same happened on chrome.

**Comment by Mankro** [comment]
@EerikSaksi the issue becomes clear when you have longer chapters and assignment descriptions. The example course contents in the video are very short and that's why the issue does not show up.

You can use the O1 course for testing this issue. For example, in this page, I can see this issue clearly:
https://plus.cs.aalto.fi/o1/2022/w02/ch05/#chapter-exercise-6
